### ISSUE TITLE
get_elev_raster(raster, z = 7, src = "aws") returns Error: API did not return tif

### ISSUE THREAD

**Issue by yipcma** [USEPA/elevatr#17]

**Comment by jhollist** [comment]
Thanks for the issue @yipcma .  Can you provide a bit more details on this?  With my own rasters I am able to get a returned raster using z=7 and src = "aws".

What are the details of the raster you are using?

**Comment by yipcma** [comment]
```
class       : RasterLayer 
dimensions  : 498, 614, 305772  (nrow, ncol, ncell)
resolution  : 0.04497312, 0.04497315  (x, y)
extent      : 32.23163, 59.84513, 12.10721, 34.50384  (xmin, xmax, ymin, ymax)
coord. ref. : +proj=longlat +datum=WGS84 +ellps=WGS84 +towgs84=0,0,0 
data source : in memory
names       : pmax 
values      : 907.3095, 2108.108  (min, max)
```
z = 7 not working on either aws or mapzen

**Comment by yipcma** [comment]
```
class       : RasterLayer 
dimensions  : 498, 614, 305772  (nrow, ncol, ncell)
resolution  : 0.04497312, 0.04497315  (x, y)
extent      : 32.23163, 59.84513, 12.10721, 34.50384  (xmin, xmax, ymin, ymax)
coord. ref. : +proj=longlat +datum=WGS84 +ellps=WGS84 +towgs84=0,0,0 
data source : in memory
names       : pmax 
values      : 907.3095, 2108.108  (min, max)
```
z = 7 not working on either aws or mapzen

**Comment by yipcma** [comment]
```
class       : RasterLayer 
dimensions  : 498, 614, 305772  (nrow, ncol, ncell)
resolution  : 0.04497312, 0.04497315  (x, y)
extent      : 32.23163, 59.84513, 12.10721, 34.50384  (xmin, xmax, ymin, ymax)
coord. ref. : +proj=longlat +datum=WGS84 +ellps=WGS84 +towgs84=0,0,0 
data source : in memory
names       : pmax 
values      : 907.3095, 2108.108  (min, max)
```
z = 7 not working on either aws or mapzen

**Comment by yipcma** [comment]
class       : RasterLayer 
dimensions  : 498, 614, 305772  (nrow, ncol, ncell)
resolution  : 0.04497312, 0.04497315  (x, y)
extent      : 32.23163, 59.84513, 12.10721, 34.50384  (xmin, xmax, ymin, ymax)
coord. ref. : +proj=longlat +datum=WGS84 +ellps=WGS84 +towgs84=0,0,0 
data source : in memory
names       : pmax 
values      : 907.3095, 2108.108  (min, max)

z = 7 not working on either aws or mapzen

**Comment by yipcma** [comment]
class       : RasterLayer 
dimensions  : 498, 614, 305772  (nrow, ncol, ncell)
resolution  : 0.04497312, 0.04497315  (x, y)
extent      : 32.23163, 59.84513, 12.10721, 34.50384  (xmin, xmax, ymin, ymax)
coord. ref. : +proj=longlat +datum=WGS84 +ellps=WGS84 +towgs84=0,0,0 
data source : in memory
names       : pmax 
values      : 907.3095, 2108.108  (min, max)

z = 7 not working on either aws or mapzen

**Comment by jhollist** [comment]
I can recreate the issue.  And I need to dig into it more, but I suspect it is an API rate limit.  That is a lot of data for a single request.  If you need the elevation at the higher resolution then you'll need to ask for that data with separate input spatial data (could create 2 (or more) squares that cover your area of interest with some overlap.  Pull down the individual rasters and mosaic them together.

When I get some more time I will look into the API restrictions on this and at least add a more informative error message for instances like this.

**Comment by TomBor** [comment]
Hi, same problem with `src = "mapzen"` with a script that worked ten days ago and today not anymore.
And [Mapzen API as no rate limit](https://mapzen.com/documentation/overview/rate-limits/#mapzen-terrain-tiles):
>Mapzen Terrain Tiles provides global elevation coverage and there are no rate limits.

**Comment by jhollist** [comment]
Hmmm....

I'll move this up my list and see I can figure out what is going on.  I am able to recreate this with the example on get_elev_raster so something appears to have happened on the Mapzen side.  

Using "aws" for the source worked on my examples so in the meantime try switching to that.

**Comment by jslingsby** [comment]
Sorry to add - Looks like src = "aws" on the example now returns the same error too...

**Comment by jhollist** [comment]
All, thanks so much for reporting this.  Turns out the issue was both AWS and Mapzen changed the spelling of tiff in the HTTP response.  I had a conditional looking for the old spelling that was stopping the function.  Fix is pushed to this repo, so try devtools::install_github("jhollist/elevatr") for now.  I hope to get the fix pushed up to CRAN in the next couple of days.

**Comment by jslingsby** [comment]
Thanks! Worked for me!

**Comment by mdsumner** [comment]
Oh neat, that's a classic kind of error - easy, but takes a while to flush!

**Comment by TomBor** [comment]
Thanks!

**Comment by jhollist** [comment]
fixed with #18 and my push.

**Comment by jhollist** [comment]
Quick update for everyone (and for https://github.com/USEPA/elevatr/issues/2).

Current version on GitHub is CRAN ready.  Just waiting for Jan 3 for CRAN to open submissions again and will submit.  Sorry all for the delay in getting this fix up.  elevatr on CRAN is pretty much useless without it!  Remedy coming soon.

Thanks again for your help in ferreting this problem out.
